tor/add_onion.go

```diff
@@ -105,8 +111,12 @@ type AddOnionConfig struct {
 
 // AddOnion creates an onion service and returns its onion address. Once
 // created, the new onion service will remain active until the connection
-// between the controller and the Tor server is closed.
-func (c *Controller) AddOnion(cfg AddOnionConfig) (*OnionAddr, error) {
+// between the controller and the Tor server is closed. This can optionally
+// take an existing private key as input and return the generated onion
+// private key.
+func (c *Controller) AddOnion(cfg AddOnionConfig, privateKeyInput string,
```

Comment:
I feel like the extra arguments could be replaced with something like this so the argument list doesn't grow too long https://github.com/lightningnetwork/lnd/blob/652ac7fa3f84f95dbe2617c9efe9e9117126a4fe/channeldb/db.go#L226